### PR TITLE
Client auth rpt

### PR DESCRIPTION
**Pull request by altexy**

### CODE REVIEW

File: kong/plugins/gluu-uma-pep/schema.lua [hunk 1]
```diff
@@ -7,7 +7,8 @@ return {
         oxd_id = { required = true, type = "string" },
         op_url = { required = true, type = "url" },
         uma_scope_expression = { required = true, type = "table" }, --TODO check structure
-        allow_unprotected_path = { type = "boolean", default = false },
+        deny_by_default = { type = "boolean", default = true },
+        anonymous = { type = "string", func = check_user, default = "" },
```
Comment (ldeveloperl1985) [inline kong/plugins/gluu-uma-pep/schema.lua]: Please add check_user function

File: kong/plugins/gluu-uma-pep/access.lua [hunk 2]
```diff
@@ -139,32 +141,56 @@ local function try_introspect_rpt(conf, token)
...
-    kong.log.debug("Hide authorization header")
-    kong.service.request.clear_header("authorization")
+    kong.service.request.set_headers(new_headers)
 end
 
+local function set_anonymous(conf)
+    print(conf.anonymous)
+    local consumer_cache_key = kong.db.consumers:cache_key(conf.anonymous)
+    local consumer, err = kong.cache:get(consumer_cache_key, nil, load_consumer_by_id, conf.anonymous)
+
```
Comment (ldeveloperl1985) [inline kong/plugins/gluu-uma-pep/access.lua]: load_consumer_by_id function is missing please add it

**Comment by ldeveloperl1985** [comment]
Also please add this functionality https://github.com/GluuFederation/gluu-gateway/issues/261

**Comment by ldeveloperl1985** [comment]
@altexy  Update request for `oxd.get_client_token` in plugins. `scope` requires array with `oxd` and `openid` scope.

**Comment by altexy** [comment]
Closed, new PR #264

**Pull request closed without merging**